part_1/src/com/chernova/library/Book.java

```diff
@@ -0,0 +1,87 @@
+package com.chernova.library;
+
+import java.util.Comparator;
+import java.util.Date;
+
+public class Book implements Comparable<Book>, Comparator<Book> {
+    private int id;
+    private String title;
+    private String genre;
+    public Date date;
+
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    public void setGenre(String genre) {
+        this.genre = genre;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getGenre() {
+        return genre;
+    }
+
+
+    public Book() {
```

Comment:
Let's add ID generation into constructor (every new book should have id greater than previously created on 1)